VSO CSI driver documentation (#691)
hashicorp/web-unified-docs: content/vault/v1.20.x/content/docs/deploy/kubernetes/index.mdx

Author: I think I'll need your review on this file @stevealmyHC to determine if modifying this existing page is sufficient for satisfying the comparison ticket for now, or if you want to go further and create a dedicated page for the comparisons. (This is currently just on the "Kubernetes" overview page.)

Author: Discussed offline, sounds like we will add a more detailed comparison as well.

### High level comparison of integrations

There are currently 4 different integrations to help Kubernetes workloads seamlessly consume secrets from Vault, without the need to modify the application to interact directly with Vault. Each integration addresses slightly different use-cases. The following is a brief overview of the strengths of each integration.

#### Agent injector

> ([docs](/vault/docs/platform/k8s/injector))

Reviewer: ```suggestion

([Agent injector docs](/vault/docs/platform/k8s/injector))
```
Style correction: use meaningful anchor text

- No durable secret storage outside Vault. All secrets written to disk are in ephemeral in-memory volumes.
- No highly privileged service accounts required. All secrets are fetched with the pod's own service account without the need for any other service accounts to impersonate it.
- Wider array of auth methods

#### Vault Secrets Operator

> ([docs](/vault/docs/platform/k8s/vso))

Reviewer: ```suggestion

([Vault Secrets Operator docs](/vault/docs/platform/k8s/vso))
```

- More native UX for app developers. Workloads can mount Kubernetes secrets without adding any Vault-specific configuration.
- Reduced load on Vault. Secrets are synced per CRD instead of per consuming pod.
- Better Vault secret availability. Kubernetes secrets act as a durable cluster-local cache of Vault secrets.

> - Ability to use templating through [SecretTransformations](/vault/docs/platform/k8s/vso/secret-transformation) to deliver secrets in the desired format

Reviewer: ```suggestion
- Supports templating with [SecretTransformations](/vault/docs/platform/k8s/vso/secret-transformation)
  that deliver secrets in the desired format.
```
Style correction: write in active voice

#### Vault Secrets Operator CSI driver

> ([docs](/vault/docs/platform/k8s/vso/vso-csi))

Reviewer: ```suggestion

([CSI driver for Vault Secrets Operator docs](/vault/docs/platform/k8s/vso/vso-csi))
```

> - All secrets are written only to ephemeral volumes, for users who wish to avoid the locked-down API server permissions and etcd encryption requirements of Kubernetes Secrets.

Reviewer: ```suggestion
- Writes all secrets to ephemeral volumes to avoid the Kubernetes Secrets
  requirements for locked-down API server permissions and `etcd` encryption.
```
Style correction: write in active voice

> - Reduced load on Vault compared to Agent Injector, as the number of connections to Vault corresponds to the number of nodes, not the number of consuming pods.

Reviewer: ```suggestion
- Associates Vault connections with nodes instead of consuming pods, which
  reduces your Vault load when compared to Agent Injector.
```

> - Ability to use templating through [SecretTransformations](/vault/docs/platform/k8s/vso/secret-transformation) to deliver secrets in the desired format

Reviewer: ```suggestion
- Supports templating with [SecretTransformations](/vault/docs/platform/k8s/vso/secret-transformation)
  that deliver secrets in the desired format.
```

#### Vault CSI provider for Secrets Store CSI driver

> ([docs](/vault/docs/platform/k8s/csi))

Reviewer: ```suggestion

([Vault CSI provider for Secrets Store CSI driver docs](/vault/docs/platform/k8s/csi))
```

> ### High level comparison of integrations

Reviewer: This section duplicates the information in the new comparisons page so we should delete it
